### PR TITLE
HIVE-25755: TestTxnCommands classes refactor

### PR DESCRIPTION
**Pull request by deniskuzZ**

### CODE REVIEW

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java [hunk 1]
```diff
@@ -1352,7 +1351,7 @@ public void run() {
       }
     }, 5000);
     long start = System.currentTimeMillis();
-    runStatementOnDriver("alter table "+ TestTxnCommands2.Table.ACIDTBL +" compact 'major' AND WAIT");
+    runStatementOnDriver("alter table "+ Table.ACIDTBL +" compact 'major' AND WAIT");
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java:1354]: nit: Maybe OCD but put spaces around the `+` signs please

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java [hunk 2]
```diff
@@ -1397,7 +1396,7 @@ public void testNonAcidToAcidConversion01() throws Exception {
         BucketCodec.V1.encode(new AcidOutputFormat.Options(hiveConf).bucket(1)));
 
     //run Compaction
-    runStatementOnDriver("alter table "+ TestTxnCommands2.Table.NONACIDORCTBL +" compact 'major'");
+    runStatementOnDriver("alter table "+ Table.NONACIDORCTBL +" compact 'major'");
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java:1399]: nit: Maybe OCD but put spaces around the `+` signs please

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java [hunk 3]
```diff
@@ -1010,7 +952,7 @@ public void testInsertOverwriteWithSelfJoin() throws Exception {
     Assert.assertEquals("Insert overwrite partition failed", stringifyValues(updatedData), rs2);
     //insert overwrite not supported for ACID tables
   }
-  private static void checkCompactionState(CompactionsByState expected, CompactionsByState actual) {
+  private void checkCompactionState(CompactionsByState expected, CompactionsByState actual) {
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:955]: nit: new line before

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java [hunk 4]
```diff
@@ -1172,7 +1114,7 @@ private int getFileSystemCacheSize() throws Exception {
       this.total = total;
     }
   }
-  private static CompactionsByState countCompacts(TxnStore txnHandler) throws MetaException {
+  private CompactionsByState countCompacts(TxnStore txnHandler) throws MetaException {
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:1117]: nit: new line before

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java [hunk 5]
```diff
@@ -767,7 +709,7 @@ public void testNonAcidToAcidConversion3() throws Exception {
     // 4. Update the existing row, and insert another row to newly-converted ACID table
     runStatementOnDriver("update " + Table.NONACIDORCTBL + " set b=3 where a=1");
     runStatementOnDriver("insert into " + Table.NONACIDORCTBL + "(a,b) values(3,4)");
-    status = fs.listStatus(new Path(TEST_WAREHOUSE_DIR + "/" +
+    status = fs.listStatus(new Path(getWarehouseDir() + "/" +
```
Comment (pvary) [inline ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:712]: Question: Would it worth to create a method for this?
`listStatus(String tableName)`

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java (hunk 2 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java]: fixed

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java (hunk 3 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:956]: fixed

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java (hunk 4 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:1119]: fixed

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java (hunk 5 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:712]: I would keep the existing signature for now, it's more generic, allows to work with partitioned tables + provide diff filters.

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java (hunk 3 above)
Comment (pvary) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java:956]: Why did we remove `static`? I usually try to keep the methods static, so it shows that the class attributes are not used

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java (hunk 1 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands.java]: fixed

File: ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java (hunk 3 above)
Comment (deniskuzZ) [reply ql/src/test/org/apache/hadoop/hive/ql/TestTxnCommands2.java]: restored

**Review by pvary — APPROVED** [review]

**Pull request merged**